protocol/kafka_confluent/v2/protocol.go

```diff
@@ -0,0 +1,212 @@
+/*
+ Copyright 2023 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_confluent
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+
+	cecontext "github.com/cloudevents/sdk-go/v2/context"
+)
+
+var (
+	_ protocol.Sender   = (*Protocol)(nil)
+	_ protocol.Opener   = (*Protocol)(nil)
+	_ protocol.Receiver = (*Protocol)(nil)
+	_ protocol.Closer   = (*Protocol)(nil)
+)
+
+type Protocol struct {
+	kafkaConfigMap *kafka.ConfigMap
+
+	consumer             *kafka.Consumer
+	consumerTopics       []string
+	consumerRebalanceCb  kafka.RebalanceCb     // optional
+	consumerPollTimeout  int                   // optional
+	consumerErrorHandler func(err kafka.Error) //optional
+	consumerMux          sync.Mutex
+
+	producer             *kafka.Producer
+	producerDeliveryChan chan kafka.Event // optional
+	producerDefaultTopic string           // optional
+
+	// receiver
+	incoming chan *kafka.Message
+}
+
+func New(opts ...Option) (*Protocol, error) {
+	p := &Protocol{
+		consumerPollTimeout: 100,
+		incoming:            make(chan *kafka.Message),
+	}
+	if err := p.applyOptions(opts...); err != nil {
+		return nil, err
+	}
+
+	if p.kafkaConfigMap != nil {
+		if p.consumerTopics != nil && p.consumer == nil {
+			consumer, err := kafka.NewConsumer(p.kafkaConfigMap)
+			if err != nil {
+				return nil, err
+			}
+			p.consumer = consumer
+		}
+		if p.producerDefaultTopic != "" && p.producer == nil {
+			producer, err := kafka.NewProducer(p.kafkaConfigMap)
+			if err != nil {
+				return nil, err
+			}
+			p.producer = producer
+		}
+		if p.producer == nil && p.consumer == nil {
+			return nil, fmt.Errorf("at least set one of the receiver or sender topic")
+		}
+	}
+
+	if p.kafkaConfigMap == nil && p.producer == nil && p.consumer == nil {
+		return nil, fmt.Errorf("at least one of the following to initialize the protocol: config, producer, or consumer")
+	}
+	if p.producer == nil {
+		p.producerDeliveryChan = make(chan kafka.Event)
+	}
+	return p, nil
+}
+
+func (p *Protocol) applyOptions(opts ...Option) error {
+	for _, fn := range opts {
+		if err := fn(p); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *Protocol) Send(ctx context.Context, in binding.Message, transformers ...binding.Transformer) (err error) {
+	if p.producer == nil {
+		return fmt.Errorf("the producer client must not be nil")
```

Comment:
nit (same for others:) 

```suggestion
		return fmt.Errorf("producer client must be set")
```